src/fiatExchanges/SelectProvider.tsx

```diff
@@ -0,0 +1,304 @@
+  const dispatch = useDispatch()
+  const userLocation = useSelector(userLocationDataSelector)
+  const account = useSelector(currentAccountSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const [noPaymentMethods, setNoPaymentMethods] = useState(false)
+  const { flow } = route.params
+
+  const digitalAsset = {
+    [Currency.Celo]: CiCoCurrency.CELO,
+    [Currency.Dollar]: CiCoCurrency.CUSD,
+    [Currency.Euro]: CiCoCurrency.CEUR,
+  }[route.params.selectedCrypto]
+
+  const asyncProviders = useAsync(async () => {
+    if (!account) {
+      Logger.error(TAG, 'No account set')
+      return
+    }
+    try {
+      const providers = await fetchProviders({
+        userLocation,
+        walletAddress: account,
+        fiatCurrency: localCurrency,
+        digitalAsset,
+        fiatAmount: route.params.amount.fiat,
+        digitalAssetAmount: route.params.amount.crypto,
+        txType: flow === CICOFlow.CashIn ? 'buy' : 'sell',
+      })
+
+      const rawLegacyMobileMoneyProviders = await fetchLegacyMobileMoneyProviders()
+      const legacyMobileMoneyProviders = filterLegacyMobileMoneyProviders(
+        rawLegacyMobileMoneyProviders,
+        flow,
+        userLocation.countryCodeAlpha2,
+        digitalAsset
+      )
+
+      return { providers, legacyMobileMoneyProviders }
+    } catch (error) {
+      dispatch(showError(ErrorMessages.PROVIDER_FETCH_FAILED))
+    }
+  }, [])
+
+  const quoteOnPress = ({ quote, provider }: CicoQuote) => () => {
+    ValoraAnalytics.track(FiatExchangeEvents.provider_chosen, {
+      flow,
+      provider: provider.name,
+    })
+
+    if (quote && userLocation?.ipAddress && isSimplexQuote(quote)) {
+      navigate(Screens.Simplex, {
+        simplexQuote: quote,
+        userIpAddress: userLocation.ipAddress,
+      })
+      return
+    }
+
+    ;(quote as ProviderQuote).url && navigateToURI((quote as ProviderQuote).url)
+  }
+
+  if (asyncProviders.loading) {
+    return (
+      <View style={styles.activityIndicatorContainer}>
+        <ActivityIndicator size="large" color={colors.greenBrand} />
+      </View>
+    )
+  }
+  const activeProviders = asyncProviders.result?.providers
+
+  const cicoQuotes: CicoQuote[] =
+    getQuotes(activeProviders)
+      ?.filter(({ quote }) => (flow === CICOFlow.CashIn ? quote.cashIn : quote.cashOut))
+      .sort(sortQuotes) || []
+  return (
+    <ScrollView>
+      <PaymentMethodSection
+        cicoQuotes={cicoQuotes}
+        paymentMethod={PaymentMethod.Card}
+        setNoPaymentMethods={setNoPaymentMethods}
+        quoteOnPress={quoteOnPress}
+      />
+      <PaymentMethodSection
+        cicoQuotes={cicoQuotes}
+        paymentMethod={PaymentMethod.Bank}
+        setNoPaymentMethods={setNoPaymentMethods}
+        quoteOnPress={quoteOnPress}
+      />
+      <LegacyMobileMoneySection
+        provider={asyncProviders.result?.legacyMobileMoneyProviders[0]}
```

Comment:
what if there are multiple legacy mobile money providers?

Reply:
Its only two and each are only available in different countries. They haven't been touched in over a year and I'm hedging my bets that we won't be adding more

Reply:
Its not that simple. The mobile money section will need to be similar to the Bank/Card sections showing an expandable dropdown if there are multiple. My reasons for not doing it in this PR include:
* The PR was already getting huge, and its not something we need yet.
* The best way to do it would probably be to make [PaymentMethodSection](https://github.com/valora-inc/wallet/blob/select-provider-squash/src/fiatExchanges/PaymentMethodSection.tsx) more flexible, but I'd like to wait until after [Wallet call fetchFiatConnectProviders and display](https://app.zenhub.com/workspaces/acquisition-squad-sprint-board-6010683afabec1001a090887/issues/valora-inc/wallet/2332) before trying to reconcile legacyMobileMoney data structure with quotes
* The next mobile money provider we add will likely be fiatconnect and so that will require wallet changes anyways

I'll bring the decision to only use the first legacyMobileMoneyProvider into the component and add a comment to clarify so that its less likely that this will be overlooked and taken as a pattern

Reply:
right, the expandable thing. Ok, makes sense. If the assumption is made more explicit I think I'm fine with it